### PR TITLE
Update global concurrency docs to be more global

### PR DESCRIPTION
**Pull request by zhen0**
The introduction to the global concurrency docs made me think that they are only for limiting task run concurrency when in fact they can be used more globally.  This is a small change to make it more clear they can be used for limiting concurrency for flow runs and other operations. 

@bunchesofdonald - I'd love your review to make sure I am not misrepresenting anything here! 

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: docs/guides/global-concurrency-limits.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ search:
 
 # Global concurrency limits and rate limits
 
-Global concurrency limits allow you to manage task execution efficiently, controlling how many tasks can run simultaneously. They are ideal when optimizing resource usage, preventing bottlenecks, and customizing task execution are priorities.
+Global concurrency limits allow you to manage execution efficiently, controlling how many tasks, flows or other operations can run simultaneously. They are ideal when optimizing resource usage, preventing bottlenecks, and customizing task execution are priorities.
```
Comment (discdiver) [inline docs/guides/global-concurrency-limits.md:12]:
```suggestion
Global concurrency limits allow you to manage execution efficiently, controlling how many tasks, flows, or other operations can run simultaneously. They are ideal when optimizing resource usage, preventing bottlenecks, and customizing task execution are priorities.
```

Comment (discdiver) [reply docs/guides/global-concurrency-limits.md:12]: Just added oxford comma for consistency with other docs

**Review by discdiver — APPROVED** [review]
Thank you!  One minor nit.

**Review by discdiver — APPROVED** [review]
Thanks!

**Review by bunchesofdonald — APPROVED** [review]
Makes sense to me, thanks @zhen0 !

**Pull request merged**